packages/core/src/modules/routing/RecipientModule.ts

```diff
@@ -178,6 +183,66 @@ export class RecipientModule {
     return event.payload.mediationRecord
   }
 
+  public async provision(mediatorConnInvite: string) {
+    this.logger.debug('Provision Mediation with invitation', { invite: mediatorConnInvite })
+    if (!mediatorConnInvite) {
+      throw new AriesFrameworkError('Mediation provision requires an invitation to be passed')
+    }
+    // Connect to mediator through provided invitation if provided in config
+    // Also requests mediation ans sets as default mediator
+    // Because this requires the connections module, we do this in the agent constructor
+
+    // Assumption: processInvitation is a URL-encoded invitation
+    const invitation = await ConnectionInvitationMessage.fromUrl(mediatorConnInvite)
+    // Check if invitation has been used already
+    if (!invitation || !invitation.recipientKeys || !invitation.recipientKeys[0]) {
+      throw new Error(`Invalid mediation invitation`)
+    }
+    const connection = await this.connectionService.findByInvitationKey(invitation.recipientKeys[0])
+    if (!connection) {
+      this.logger.debug('Mediation Connection does not exist, creating connection')
+      const routing = await this.mediationRecipientService.getRouting()
+
+      const invitationConnectionRecord = await this.connectionService.processInvitation(invitation, {
+        autoAcceptConnection: true,
+        routing: routing,
+      })
+      this.logger.debug('Processed invitation')
+      const { message, connectionRecord: connectionRecord } = await this.connectionService.createRequest(
+        invitationConnectionRecord.id
+      )
+      const outbound = createOutboundMessage(connectionRecord, message)
```

Comment:
We should find a way around this as we're exactly duplicating the connection module code here (not for this PR, just something to think about)

Reply:
Do we want to be able to import the module into a module? I don't think right now the modules are set up to be used via dependency injection IIRC.

Reply:
No as we've set it up, modules don't use modules. But it has introduced some limitations...